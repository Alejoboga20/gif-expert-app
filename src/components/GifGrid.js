import React from 'react';

const GifGrid = ({ category }) => {
  const getGif = async () => {
    const url =
      'https://api.giphy.com/v1/gifs/search?q=React&limit=10&api_key=U81A19qAgRSt2BXfoSr6MvmVr1OyqhNP';

    const response = await fetch(url);
    const { data } = await response.json();
    const gifs = data.map((img) => {
      return {
        id: img.id,
        title: img.title,
        url: img.images.downsized_medium.url
      };
    });
    console.log(gifs);
  };

  getGif();

  return (
    <div>
      <h3>{category}</h3>
    </div>
  );
};

export default GifGrid;
