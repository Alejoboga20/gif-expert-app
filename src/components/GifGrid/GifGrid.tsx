import { Loading, Grid, Text } from '@nextui-org/react';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import GifGridItem from '../GifGridItem';

const GifGrid = ({ category }: GifGridProps) => {
	const { data: images, loading } = useFetchGifs(category);

	if (loading) return <Loading size='lg' />;

	return (
		<>
			<Text h3>{category}</Text>

			<Grid.Container gap={1} justify='center'>
				{images.map((image) => (
					<Grid xs={6} sm={4} md={3} lg={2}>
						<GifGridItem key={image.id} {...image} />
					</Grid>
				))}
			</Grid.Container>
		</>
	);
};

interface GifGridProps {
	category: string;
}

export default GifGrid;
