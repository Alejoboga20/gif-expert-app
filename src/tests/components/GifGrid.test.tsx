import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import GifGrid from '../../components/GifGrid/GifGrid';

jest.mock('../../hooks/useFetchGifs');
const mockedUseFetchGifs = useFetchGifs as jest.MockedFunction<typeof useFetchGifs>;

const gifs = [
	{
		id: 'ABC',
		url: 'https://localhost.com',
		title: 'Testing',
	},
];

describe('GifGrid Tests', () => {
	const category = 'Avatar';

	test('should be render properly when is loading', () => {
		mockedUseFetchGifs.mockReturnValue({ data: [], loading: true });
		const wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
	});

	test('should show items when images are loaded with useFetchGifs', () => {
		mockedUseFetchGifs.mockReturnValue({ data: gifs, loading: false });
		const wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
	});
});
