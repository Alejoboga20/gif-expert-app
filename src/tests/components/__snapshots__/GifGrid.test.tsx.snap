// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid Tests should be render properly when is loading 1`] = `
<Memo(Loading)
  size="lg"
  type="default"
/>
`;

exports[`GifGrid Tests should show items when images are loaded with useFetchGifs 1`] = `
<Fragment>
  <Memo(Text)
    b={false}
    blockquote={false}
    color="default"
    del={false}
    em={false}
    h1={false}
    h2={false}
    h3={true}
    h4={false}
    h5={false}
    h6={false}
    i={false}
    small={false}
    span={false}
    transform="none"
  >
    Avatar
  </Memo(Text)>
  <GridContainer
    className=""
    gap={1}
    justify="center"
    wrap="wrap"
  >
    <Grid
      key=".0"
      lg={2}
      md={3}
      sm={4}
      xs={6}
    >
      <GifGridItem
        id="ABC"
        key="ABC"
        title="Testing"
        url="https://localhost.com"
      />
    </Grid>
  </GridContainer>
</Fragment>
`;
